### DISCUSSION TITLE
Will reanimated 2 have v8 support?

### DISCUSSION THREAD

**Discussion by shamilovtim** [software-mansion/react-native-reanimated#1897] [Ideas]
Just wondering if you plan to support the v8 engine? AFAIK it's gaining popularity due to outperforming Hermes on Android.

**Comment by andreialecu** [comment]
There's another discussion about v8 here: https://github.com/software-mansion/react-native-reanimated/discussions/1913

**Comment by mrousavy** [comment]
Krzysztof's cooking something up in https://github.com/software-mansion/react-native-reanimated/pull/1960 🔥

**Comment by jzxchiang1** [comment]
Bump. react-native-v8 is now compatible with reanimated v2 (I tested it myself with the help of @Kudo).

Can we please get official support from reanimated as soon as possible? Thank you!

https://github.com/software-mansion/react-native-reanimated/pull/1960#issuecomment-940043904

**Comment by shamilovtim** [comment]
Bump. Is there a v8 cut in the works?

**Reply by jzxchiang1** [reply]
I hope so! v8 already fixed the blocking [issue](https://github.com/Kudo/react-native-v8/commit/a82991d5e841a7fe68f8fbbae728e388cce776b5) on their side.

**Comment by jzxchiang1** [comment]
For those waiting on official support, you can run

```
yarn add react-native-reanimated@npm:react-native-reanimated-v8@2.2.2
```

to get a version of reanimated that's compatible with v8. I've been using it and it works beautifully. The only issue is that it doesn't work with other libraries with native dependencies on reanimated, for example [react-native-vision-camera](https://github.com/mrousavy/react-native-vision-camera/issues/499#issuecomment-943245994) by @mrousavy. That's because it depends on a properly configured / named reanimated AAR for v8.

I do hope reanimated can officially support v8 soon though, since all the pieces are there. The work has in fact already been done by @Kudo [here](https://github.com/Kudo/react-native-reanimated/commit/4b069950c4b75b836350b1803485819939ce632b); it's mostly modifying some build scripts.

Would make a big difference for v8 users, thanks!

**Comment by DrRefactor** [comment]
Are there any plans to integrate @Kudo's work regarding v8 support to reanimated repo?
@Kudo maybe you could create PR from your fork?

**Comment by shamilovtim** [comment]
Bump

**Comment by jzxchiang1** [comment]
Bump! AFAIK v8 outperforms JSC (which is [outdated](https://github.com/react-native-community/jsc-android-buildscripts/releases/tag/v250230.2.1) anyways) on almost all major benchmarks on Android. JIT v8 + reanimated v2 seems like a formula for great UXs.

**Reply by mrousavy** [reply]
Why are you comparing against JSC? Hermes is the runtime you should be comparing against, that's what RN's future is.

**Reply by jzxchiang1** [reply]
Because Hermes has even fewer ECMAScript features that I need, most notably BigInt and WebAssembly.

**Comment by jzxchiang1** [comment]
Sorry to bother, but do you guys have any plans to officially support react-native-v8?

All of the foundations are there—all you have to do is take the last two commits [here](https://github.com/lkinasiewicz/react-native-reanimated/commits/v8/2.3.1) and port them over into this codebase.

It would mean a lot to v8 users, which are plenty on Android. 🙏

**Reply by Kudo** [reply]
@jzxchiang1 as the latest reanimated support to build from source, it would be a fair approach to support v8 from reanimated. i'll create a pr when a get a chance.

**Reply by jzxchiang1** [reply]
Awesome! I thought reanimated always supported building from source from a developer's machine. Didn't know something had changed.

**Reply by jzxchiang1** [reply]
Let me know if you need any help!

**Comment by jzxchiang1** [comment]
@[piaskowyk](https://github.com/piaskowyk)

**Comment by Kudo** [comment]
finally to get the integration working. open a pr at https://github.com/software-mansion/react-native-reanimated/pull/3132. this requires `react-native-v8>=1.1.0` and `react-native>=0.66.0`

**Reply by jzxchiang1** [reply]
Very much appreciated, thank you! It seems like you tested it on 0.67 but not 0.66 though.
